src/sc2api/sc2_proto_to_pods.cc

```diff
@@ -202,9 +207,15 @@ bool Convert(const ObservationRawPtr& observation_raw, UnitPool& unit_pool, uint
         unit->is_on_screen = observation_unit.is_on_screen();
         unit->is_blip = observation_unit.is_blip();
 
-        unit->health = observation_unit.health();
+        const auto health = observation_unit.health();
+        if (health < unit->health)
+            unit_pool.AddUnitDamaged(unit);
+        unit->health = health;
         unit->health_max = observation_unit.health_max();
         unit->shield = observation_unit.shield();
+        const auto shield = observation_unit.shield();
+        if (shield < unit->shield)
+            unit_pool.AddUnitDamaged(unit);
```

Comment:
Another question here: how often such events are issued?

Reply:
once for every observation in which a unit takes damage. It is often. I have a batch version of this that sends out the vector of all friendly damaged units. for my own bot, I would rather have a map for is_damaged[unit *]. I think of using the notification to create such a map. But I'm not sure in the end, if that is the best approach. The whole notification system might be better implemented as a pull instead of a push. Maybe with a multi-index (ala boost) that allows users to setup indices that are important.

Reply:
I don't want to bring in boost as it usually causes additional difficulties for less experienced programmers. While it is relatively easy to install proper boost version on Linux and OS X, Windows is usually pain. So, unless we have very strong necessity, better to avoid usage of boost in this project.

Reply:
yeah, maybe best not to complicate things.